app/gui/src/controller/ide/desktop.rs

```diff
@@ -164,3 +161,27 @@ impl ManagingProjectAPI for Handle {
         .boxed_local()
     }
 }
+
+/// Select a new name for the project in a form of <suggested_name>_N, where N is a unique sequence
+/// number.
+fn choose_new_project_name(
+    mut existing_names: impl Iterator<Item = String>,
+    suggested_name: &str,
+) -> String {
+    let with_suffix = (1..).map(|i| format!("{}_{}", suggested_name, i));
+    let mut candidates = std::iter::once(suggested_name.to_owned()).chain(with_suffix);
```

Comment:
I don't think this should work. Was it tested? 
`Foo_1` is not a valid project name, as each segment must start with an uppercase letter. `1` is not one.

Reply:
> Foo_1 is not a valid project name, as each segment must start with an uppercase letter. 1 is not one.

`Foo_1` is actually a totally valid project name. Please see [`ReferentName`](https://github.com/enso-org/enso/blob/f03640b93a242237f78e6095c9d19cc16f4cb8ce/app/gui/controller/double-representation/src/identifier.rs#L213) documentation.

Reply:
The `ReferentName` is known to be incomplete when it comes to validation: https://github.com/enso-org/enso/issues/2797
However, you are right about `Foo_1` being valid, non-first words within an identifier are allowed to be numbers.
See https://github.com/enso-org/enso/blob/develop/docs/syntax/naming.md#naming-constructs for the official source of truth about identifiers.